pkg/api/api_test.go

```diff
@@ -47,201 +47,183 @@ type testCases struct {
-		Resolvers: &resolver.Resolver{Observers: make(map[string]chan *graphql.User)},
-	}))
-	return e, graphqlHandler
-}
-
-func setupMockFunctions() {
-	ApplyFunc(boil.SetDB, mockSetDB)
-}
-
-func mockSetDB(db boil.Executor, tt testCases) {
-	fmt.Print("boil.SetDB called\n")
-	tt.setDbCalled = true
-}
-
-func hasTransportCalls(tt testCases) bool {
-	return tt.getTransportCalled || tt.postTransportCalled ||
-		tt.optionsTransportCalled || tt.multipartFormTransportCalled
-}
-
-func checkTransportCalls(tt testCases) {
-	observers := map[string]chan *graphql.User{}
-	graphqlHandler := handler.New(graphql.NewExecutableSchema(graphql.Config{
-		Resolvers: &resolver.Resolver{Observers: observers},
-	}))
-	if tt.getTransportCalled || tt.postTransportCalled ||
-		tt.optionsTransportCalled || tt.multipartFormTransportCalled {
-		ApplyMethod(reflect.TypeOf(graphqlHandler), "AddTransport", func(s *handler.Server, t graphql2.Transport) {
-			transportGET := transport.GET{}
-			transportMultipartForm := transport.MultipartForm{}
-			transportPOST := transport.POST{}
-			transportWebsocket := transport.Websocket{
-				KeepAlivePingInterval: 10 * time.Second,
-				InitFunc: func(ctx context.Context, initPayload transport.InitPayload) (context.Context, error) {
-					return ctx, nil
-				},
-				Upgrader: websocket.Upgrader{
-					CheckOrigin: func(r *http.Request) bool {
-						return true
-					},
-				},
-			}
-			if t == transportGET {
-				tt.getTransportCalled = true
-			}
-			if t == transportMultipartForm {
-				tt.multipartFormTransportCalled = true
-			}
-			if t == transportPOST {
-				tt.postTransportCalled = true
-			}
-			if reflect.TypeOf(t) == reflect.TypeOf(transportWebsocket) {
-				tt.websocketTransportCalled = true
-			}
-		})
-	}
+func checkGraphQLGetResponse(e *echo.Echo) (*http.Response, error) {
+	_, res, err := testutls.SimpleMakeRequest(testutls.RequestParameters{
+		E:          e,
+		Pathname:   "/playground",
+		HttpMethod: "GET",
+		IsGraphQL:  false,
+	})
+	return res, err
 }
-func checkGraphQLServer(t *testing.T, tt testCases, e *echo.Echo) {
-	jsonRes, err := testutls.MakeRequest(testutls.RequestParameters{
+func checkGraphQLPostResponse(e *echo.Echo) (map[string]interface{}, error) {
+	res, err := testutls.MakeRequest(testutls.RequestParameters{
 		E:           e,
 		Pathname:    graphQLPathname,
 		HttpMethod:  "POST",
 		RequestBody: testutls.MockWhitelistedQuery,
 		IsGraphQL:   false,
 	})
-	if err != nil {
-		t.Fatalf("Failed to make request to GraphQL server: %v", err)
+	return res, err
+}
+func TestStart(t *testing.T) {
+	tests := loadTestStartCases()
+	e := echo.New()
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			patches := tt.init(e, tt)
+			if tt.getTransportCalled || tt.postTransportCalled ||
+				tt.optionsTransportCalled || tt.multipartFormTransportCalled {
+				testWithTransportCalls(t, tt)
+			} else {
+				testWithoutTransportCalls(t, tt, e)
```

Comment:
lets remove these fucntions. THe assertions need to be in the Test function itself. 


```assert.Equal(t, tt.getTransportCalled, true)```

instead of hard codging the true, we should get it from `tt.want`

Reply:
this is for all assertions where this is applicable not just this one occurance pls